pkg/frontend/querymiddleware/remote_read.go

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querymiddleware
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+
+	"github.com/grafana/mimir/pkg/util"
+)
+
+func ParseRemoteReadRequestWithoutCosumingBody(req *http.Request) (url.Values, error) {
+	params := make(url.Values)
+
+	if req.Body == nil {
+		return params, nil
+	}
+
+	bodyBytes, err := util.ReadRequestBodyWithoutConsuming(req)
```

Comment:
In `querier.RemoteReadHandler` we use `util.ParseProtoReader()` which supports a max body size, in order to not blow up the memory if the request body is huge (we have an hardcoded limit of 1MB). Here we're missing such protection, and I think it's something we should do here as well.

Reply:
I think there's protection already before it https://github.com/grafana/mimir/blob/ee069d247c7c22848ee668027c432a97e120cd9f/pkg/frontend/transport/handler.go#L185  , in any case I switched to util.ParseProtoReader()

Reply:
It's not trivial to add the size limiting, so I ended up commenting the functions to say that it is the reader's responsibility.

The handler does it as far as I understand in the linked part of the code https://github.com/grafana/mimir/blob/ee069d247c7c22848ee668027c432a97e120cd9f/pkg/frontend/transport/handler.go#L185

Reply:
> I think there's protection already before it

You're absolutely right. I missed it!